GitUI/UserControls/FileStatusList.cs

```diff
@@ -358,7 +358,7 @@ private void SetFileStatusListVisibility(bool filesPresent)
         {
             LoadingFiles.Visible = false;
             FilterComboBox.Visible = filesPresent || (SearchComboBox.Visible && !string.IsNullOrEmpty(SearchComboBox.Text));
-            NoFiles.Visible = !FilterComboBox.Visible;
+            NoFiles.Visible = !filesPresent;
```

Comment:
Then rather make that `SearchComboBox.Visible` has already been set to `false` before calling this function.

Reply:
Sorry temporary push during investigation at a moment I though I get the reason but in fact not. Real fix pushed and just posted an explanation...